src/ui/hooks.js

```diff
@@ -121,3 +121,21 @@ export function useStableCallback(callback) {
   callbackRef.current = callback;
   return useCallback((...args) => callbackRef.current?.(...args), [callbackRef]);
 }
+
+export function useScrollTopOffset() {
+  const [topOffset, setTopOffset] = useState(0);
```

Comment:
Maybe we should add something like:
~~~
useEffect(() => {
  setTopOffset(document.documentElement.scrollTop);
}, []);
~~~
to set the initial position on the client side in case it's not 0?

Reply:
Good point, added!